doc/specs/stdlib_math.md

```diff
@@ -0,0 +1,101 @@
+---
+title: math
+---
+
+# The `stdlib_math` module
+
+[TOC]
+
+## Introduction
+
+`stdlib_math` module provides with general purpose mathematical functions.
+
+
+## Procedures and Methods provided
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### `clip` function
+
+#### Description
+
+Limits the input value `x` to the given interval [`xmin`, `xmax`] (interval is `xmin` and `xmax` inclusive). Returns a value which lies in the given interval and is closest to the input value `x`.  
+If the input value `x` already lies in the given interval, then the output value will be equal to the input value.
+
+Note: A valid input must **NOT** have `xmin` value greater than `xmax` value.
+
+#### Syntax
+
+`res = [[stdlib_math(module):clip(interface)]] (x, xmin, xmax)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument(s)
+
+`x`: scalar of either `integer` or `real`. This argument is `intent(in)`.  
+`xmin`: scalar of either `integer` or `real`. This argument is `intent(in)`.  
+`xmax`: scalar of either `integer` or `real`. This argument is `intent(in)`.
+
+Note: All arguments must have same `type` and same `kind`.
+
+#### Output value or Result value
+
+Output is a scalar of either `integer` or `real` depending on the arguments. The output value will have `type` and `kind` same as to that of the arguments.
```

Comment:
```suggestion
The output value is a scalar of the same `type` and `kind` as to that of the arguments.
```

Reply:
can we instead write:
The output is a scalar of type and kind same as to that of the arguments?